diskqueue.go

```diff
@@ -326,10 +340,10 @@ func (d *diskQueue) readOne() ([]byte, error) {
 	d.nextReadPos = d.readPos + totalBytes
 	d.nextReadFileNum = d.readFileNum
 
-	// TODO: each data file should embed the maxBytesPerFile
-	// as the first 8 bytes (at creation time) ensuring that
+	// use the read file exact size, ensuring that
 	// the value can change without affecting runtime
-	if d.nextReadPos > d.maxBytesPerFile {
+	// only readFileNum less than writeFileNum need move next file
+	if d.readFileNum < d.writeFileNum && d.nextReadPos >= d.curReadFileSize {
```

Comment:
I'm not sure if `d.readFileNum < d.writeFileNum` needed to be added here, I'll think on it a bit...

Adding a test or two would help :) In particular, one that fails with current master, and passes with this branch. Note that Travis-CI currently fails to post "commit status" for the "legacy travis-ci.org integration". I've updated a couple of other repos to use GitHub Actions instead, I'll get around to this one eventually. In the meantime, you can view test results at https://travis-ci.org/github/nsqio/go-diskqueue/pull_requests or run them locally.

Reply:
when `d.readFileNum == d.writeFileNum` means reader and writer is same file, so when `d.nextReadPos >= d.curReadFileSize` the readFile can't be closed, the program should be waiting writeFile update  `d.curReadFileSize`

And I think just pass now unit test will be ok.